netty/src/main/java/io/grpc/transport/netty/NettyClientTransport.java

```diff
@@ -112,7 +116,19 @@
     }
 
     handler = newHandler();
-    negotiationHandler = negotiator.newHandler(handler);
+    FailureListener closeOnNegotiationFailure = new FailureListener() {
+      @Override
+      public void negotiationFailed(ChannelHandlerContext ctx, Throwable cause) {
+        failureCause = cause;
+        ctx.close().addListener(new ChannelFutureListener() {
```

Comment:
It looks like we need the listener on closeFuture independent of whether negotiation failed. Calling ctx.close() here can be appropriate, but the listener itself should probably be added where it used to.